### ISSUE TITLE
Error: thread 'main' panicked at 'byte index 130

### ISSUE THREAD

**Issue by viridityzhu** [rossmacarthur/kb-remap#3]
I follow the readme to install kb-remap via cargo, and run `kb-remap --list`. Then the following error occurs:
```
> kb-remap --list
thread 'main' panicked at 'byte index 130 is out of bounds of `0x4d9    0xa293    0x1130000  1         6     0x10001baac USB       AppleUserHIDDevice         OBINS`', /Users/me/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kb-remap-0.3.1/src/hid.rs:59:64
```

It seems to indicate that my keyboards are not recognized correctly somehow. My keyboards are **Anne Pro 2**. And the macOS is **13.3.1 (a)**. 

I would greatly appreciate any insights or guidance on resolving this issue. Thank you for your assistance.

**Comment by rossmacarthur** [comment]
@viridityzhu thanks for the report could you run with `RUST_BACKTRACE=1` and paste that output here

```
RUST_BACKTRACE=1 kb-remap --list
```

Additionally the output of following would be useful (just the Devices: section)

```
hidutil list
```

**Comment by viridityzhu** [comment]
@rossmacarthur Thank you so much for your prompt response!
```
> RUST_BACKTRACE=1 kb-remap --list
thread 'main' panicked at 'byte index 130 is out of bounds of `0x4d9    0xa293    0x1130000  1         5     0x10001baa7 USB       AppleUserHIDDevice         OBINS`', /Users/me/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kb-remap-0.3.1/src/hid.rs:59:64
stack backtrace:
   0: rust_begin_unwind
             at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/core/src/panicking.rs:67:14
   2: core::str::slice_error_fail_rt
   3: core::str::slice_error_fail
             at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/core/src/str/mod.rs:87:9
   4: <hashbrown::map::HashMap<K,V,S,A> as core::iter::traits::collect::Extend<(K,V)>>::extend
   5: kb_remap::hid::list
   6: kb_remap::main
```

```
Devices:
VendorID ProductID LocationID UsagePage Usage RegistryID  Transport Class                      Product                            UserClass                 Built-In
0x0      0x0       0xa1       65280     11    0x1000007bc FIFO      AppleHIDTransportHIDDevice Apple Internal Keyboard / Trackpad (null)                    1
0x5ac    0x8104    0x0        32        138   0x1000003bb SPU       AppleSPUHIDDevice          (null)                             (null)                    1
0x4d9    0xa293    0x1130000  1         6     0x10001baac USB       AppleUserHIDDevice         OBINS
AnnePro2                     AppleUserUSBHostHIDDevice (null)
0x5ac    0x8104    0x0        65280     5     0x100000498 SPU       AppleSPUHIDDevice          (null)                             (null)                    1
0x0      0x0       0xa1       65280     3     0x1000007af FIFO      AppleHIDTransportHIDDevice Apple Internal Keyboard / Trackpad (null)                    1
0x4d9    0xa293    0x1130000  12        1     0x10001baa8 USB       AppleUserHIDDevice         OBINS
AnnePro2                     AppleUserUSBHostHIDDevice (null)
0x0      0x0       0x0        65280     255   0x1000003de SPU       AppleSPUHIDDevice          (null)                             (null)                    1
0x4d9    0xa293    0x1130000  1         0     0x10001baa6 USB       AppleUserHIDDevice         OBINS
AnnePro2                     AppleUserUSBHostHIDDevice (null)
0x0      0x0       0x0        12        1     0x10000063e Audio     (null)                     Headset                            (null)                    (null)
0x0      0x0       0xa1       1         2     0x1000007c9 FIFO      AppleHIDTransportHIDDevice Apple Internal Keyboard / Trackpad (null)                    1
0x5ac    0x8104    0x0        65280     9     0x1000003c3 SPU       AppleSPUHIDDevice          (null)                             (null)                    1
0x5ac    0x8104    0x0        65280     3     0x1000004e9 SPU       AppleSPUHIDDevice          (null)                             (null)                    1
0x0      0x0       0xa1       65280     13    0x1000007b3 FIFO      AppleHIDTransportHIDDevice Apple Internal Keyboard / Trackpad (null)                    1
0x0      0x0       0x0        65292     5     0x1000003b8 SPU       AppleSPUHIDDevice          (null)                             (null)                    1
0x4d9    0xa293    0x1130000  1         5     0x10001baa7 USB       AppleUserHIDDevice         OBINS
AnnePro2                     AppleUserUSBHostHIDDevice (null)
0x0      0x0       0x5        65280     4     0x100000465 SPU       AppleSPUHIDDevice          (null)                             (null)                    1
0x0      0x0       0xa1       1         6     0x1000007ad FIFO      AppleHIDTransportHIDDevice Apple Internal Keyboard / Trackpad (null)                    1
0x5ac    0x0       0x0        65280     15    0x1000008a6 USB       (null)                     Keyboard Backlight                 (null)                    1
0x0      0x0       0x0        65292     1     0x1000003c4 SPU       AppleSPUHIDDevice          (null)                             (null)                    1
0x0      0x0       0x0        65280     72    0x100000631 SPMI      AppleBTM                   BTM                                (null)                    (null)
```

**Comment by rossmacarthur** [comment]
Thanks! I think the product name has a new line character in it 🤯, so it's screwing up my parsing. Will look into this evening (UTC+2)

**Comment by viridityzhu** [comment]
Ahhh, that sounds like quite an unexpected situation indeed.
No need to rush, I really appreciate your prompt response and patient assistance. 
Thank you very much!

**Comment by rossmacarthur** [comment]
Fixed in 5f048606c0a08c165729977714e9156074a55009

**Comment by viridityzhu** [comment]
Fantastic!!
